extension/scripts/Selector/SelectorTable.js

```diff
@@ -183,41 +239,66 @@ var SelectorTable = {
                 return "thead tr:nth-of-type(" + (rowIndex + 1) + ")";
             }
         }
-        else if (firstRow.find("th:not(:empty)").length > 1) {
-            return "tr:nth-of-type(1)";
-        }
-        else if (firstRow.find("th:first-child:not(:empty)").length == 1 && firstRow.children().length > 1) {
-            return "tr";
-        }
-        else if ($table.find("tr td:not(:empty), tr th:not(:empty)").length) {
-            var $rows = $table.find("tr");
-            // first row with data
-            var rowIndex = $rows.index($rows.filter(":has(td:not(:empty)),:has(th:not(:empty))")[0]);
-            return "tr:nth-of-type(" + (rowIndex + 1) + ")";
-        }
         else {
-            return "";
+            if (!verticalTable) {
+                if (firstRow.find("th:not(:empty)").length > 1) {
+                    return "tr:nth-of-type(1)";
+                } else if (firstRow.find("th:first-child:not(:empty)").length === 1 && firstRow.children().length > 1) {
+                    return "tr";
+                } else if ($table.find("tr td:not(:empty), tr th:not(:empty)").length) {
+                    var $rows = $table.find("tr");
+                    // first row with data
+                    var rowIndex = $rows.index($rows.filter(":has(td:not(:empty)),:has(th:not(:empty))")[0]);
+                    return "tr:nth-of-type(" + (rowIndex + 1) + ")";
+                } else {
+                    return "";
+                }
+            }
+            else {
+                if (firstRow.find("th").length>0) return "tr>th";
```

Comment:
- most coding styles advise to use braces for all control structures (i.e. if, else, etc), even if the body contains only a single statement
- `.length > 0` check may be replaced with `.length` check